### ISSUE TITLE
React and React-Dom are blocked on Chrome even when running this on a Python 3.4 http.server

### ISSUE THREAD

**Issue by JamesDonnelly** [facebook/react#6283]
Pasting the exact code from https://facebook.github.io/react/downloads.html into a HTML file and opening it through a Python 3.4 HTTP server for some reason gives the following error in Chrome:

> Redirect at origin 'https://fb.me' has been blocked from loading by Cross-Origin Resource Sharing policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8000' is therefore not allowed access.

I'm able to work around this completely by removing the `integrity` and `crossorigin` attributes from the React and React-Dom script tags:

    <script src="https://fb.me/react-0.14.7.min.js"></script>
    <script src="https://fb.me/react-dom-0.14.7.min.js"></script>

I've never had this issue in Chrome before, and I frequently include external scripts with `crossorigin` and `integrity` attributes with no problem, so I'm lead to believe this may have something to do with the way FB.me is configured.

**Comment by gaearon** [comment]
I can’t find mentions of `integrity` or `crossorigin` on that page.
Can you help me understand where you see it?

**Comment by JamesDonnelly** [comment]
Ah, my mistake, this comes when copying over the `script` elements from the Downloads page: https://facebook.github.io/react/downloads.html.

**Comment by gaearon** [comment]
Thanks. I can reproduce this.
@zpao Can you give any insight into this?

**Comment by petetnt** [comment]
Something like [this Gist on enabling CORS on SimpleHTTPServer](https://gist.github.com/enjalot/2904124) might help.

**Comment by jimfb** [comment]
@petetnt The problem is that the headers aren't being set on the CDN, so the python gist is irrelevant / will not solve this issue.

This is a security feature of browsers, since checking integrity can leak information about the content of the response.

fbcdn does allow all origins, but fb.me does not, which is the root cause of the error here.

Does fb.me serve anything directly, or is always just redirects?  If it always just redirects, we might be able to blindly enable the CORS headers on fb.me without any security impact (thereby relying on the redirect-destination to control the CORS permissions).  Otherwise, we would need to enable the cross-origin headers for individual URLs, which will require adding data to the fburl database, which is probably a more intense task.

Another option is to use a different URL (not fb.me).  Could link directly to the CDN, for instance.

I think the shortest term solution is to remove the offending attributes from the script tag, which makes the tags look cleaner/simpler (read: less scary to new user) anyway.

**Comment by petetnt** [comment]
Oh okay, thanks Jim :+1:

**Comment by zpao** [comment]
fb.me is always a redirect, but I thought it was set up right such that this worked. I thought I checked the headers when I added those attributes to the docs... maybe not… I think ultimately we shouldn't use fb.me but haven't gone and done that yet. We might just set up a little mini-cdn of our own since we have some other issues with the older fb.me links (corrupted cache somewhere that we thought we tracked down but still see occasional issues). cdnjs and jsdelivr are both great external third-party options right now so I definitely recommend them over fb.me regardless. We just need some control when we ship releases so relying on them entirely isn't in the cards.

I think we should keep the attributes (when they work) - moving towards a world with these checks is a good thing and we can help promote these features. But shorter-term, yea that's fine.
